### ISSUE TITLE
Support for images with srcset?

### ISSUE THREAD

**Issue by hatsumatsu** [dompdf/dompdf#1484]
Does DOMPDF support images without a `src` attribute but with `srcset` and `sizes` aka repsonsive images?

**Comment by simonberger** [comment]
No it isn't supported.
While it shouldn't be that hard to implement I do not see much reason to do so.
In fact in current state most developers create templates and render those with dompdf.
It yet isn't capable to render complex responsive sites developed for browsers.

**Comment by hatsumatsu** [comment]
Okay I see. The use case I stumbled across this is about using the library in a CMS environment (namely Wordpress) where srcset-based images are saved in a monolithic content block.

One workaround would be converting the `srcset` attributes to `src` like

````
// get all images with srcset, take the last source in srcset and set it as src attribute
$dom = new DOMDocument();
$dom->loadHTML( $content );
$images = $dom->getElementsByTagName( 'img' );
foreach( $images as $image ) {
    if( $srcset = $image->getAttribute( 'srcset' ) ) {
        $sources = explode( ',', $srcset );
        $source = trim( $sources[ count( $sources ) - 1 ] );
        $source = explode( ' ', $source );
        $src = trim( $source[0] );

        $image->setAttribute( 'src', $src );
    }
}

$content = $dom->saveHTML();
`````

**Comment by simonberger** [comment]
Good solution.
I read about all possible configurations with srcset and sizes. To implement that completely functional it's a lot to do. A solution like you did here would be easily possible but I'm not sure if that's worth it.
Most tags should have a src attribute for backward compatibility anyway. I'm surprised that's not like this in your case.

**Comment by hatsumatsu** [comment]
That's true, there is quite a number of combinations of `srcset` and `sizes` and how legacy browsers deal with it. Add third-party libraries and polyfills (like [https://github.com/aFarkas/lazysizes](lazysizes.js)) and the number of markup variations is even higher. 

Choosing the largest image size from `srcset` might be a good idea anyway since the fallback `src` is likely to hold a smaller image size...
